lib/lightning_web/live/credential_live/oauth_component.ex

```diff
@@ -136,142 +164,243 @@ defmodule LightningWeb.CredentialLive.OauthComponent do
+              clip-rule="evenodd"
+            />
+          </svg>
+        </div>
+        <div class="ml-3 flex-1 md:flex md:justify-between">
+          <p class="text-sm text-slate-700">
+            Please re-authenticate to save your credential with the updated scopes
+          </p>
+          <p class="mt-3 text-sm md:ml-6 md:mt-0">
+            <.link
+              href={@authorize_url}
+              id="re-authorize-button"
+              target="_blank"
+              class="whitespace-nowrap font-medium text-blue-700 hover:text-blue-600"
+              phx-click="authorize_click"
+              phx-target={@myself}
+            >
+              Re-authenticate <span aria-hidden="true"> &rarr;</span>
+            </.link>
+          </p>
         </div>
-        <span class="text-xl">
-          Sign in with <%= provider_name(@provider) |> String.capitalize() %>
-        </span>
-      </div>
-      <div class="text-sm ml-1">
-        Not working?
-        <.link
-          href={@authorize_url}
-          target="_blank"
-          phx-target={@myself}
-          phx-click="authorize_click"
-          class="hover:underline text-primary-900"
-        >
-          Reauthorize.
-        </.link>
       </div>
     </div>
     """
   end
 
-  def error_block(%{type: :userinfo_failed} = assigns) do
+  def userinfo_skeleton(assigns) do
     ~H"""
-    <div class="mx-auto pt-2 max-w-md">
-      <div class="text-center">
-        <Heroicons.exclamation_triangle class="h-6 w-6 text-red-600 inline-block" />
-        <div class="text-base font-medium text-gray-900">
-          Something went wrong.
+    <div
+      id="userinfo_skeleton"
+      class="border border-slate-300 shadow rounded-md mt-5 p-4 max-w"
+    >
+      <div class="animate-pulse flex space-x-4">
+        <div class="rounded-full bg-slate-700 h-10 w-10"></div>
+        <div class="flex-1 space-y-6 py-1">
+          <div class="h-2 bg-slate-700 rounded"></div>
+          <div class="space-y-3">
+            <div class="grid grid-cols-3 gap-4">
+              <div class="h-2 bg-slate-700 rounded col-span-2"></div>
+              <div class="h-2 bg-slate-700 rounded col-span-1"></div>
+            </div>
+            <div class="h-2 bg-slate-700 rounded"></div>
+          </div>
         </div>
-        <p class="text-sm mt-2">
-          Failed retrieving your information.
-        </p>
-        <p class="text-sm mt-2">
-          Please
-          <a
-            href="#"
-            phx-click="try_userinfo_again"
-            phx-target={@myself}
-            class="hover:underline text-primary-900"
-          >
-            try again.
-          </a>
-        </p>
-        <.helpblock provider={@provider} type={@type} />
       </div>
     </div>
     """
   end
```

Comment:
This looks a bit off when loading.

Reply:
![image](https://github.com/OpenFn/lightning/assets/144796/5f3992fc-388c-4b04-b4d8-2d8d206ef0c0)